Sources/RestKit/Authentication/ConfigBasedAuthenticatorFactory.swift

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright IBM Corporation 2019
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import Foundation
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+
+public enum AuthenticatorError: Error {
+    case noConfigurationFound
+    case authenticationTypeNotDefined
+    case authenticationTypeNotRecognized
+    case missingEnvironmentVariable(String)
+}
+
+public enum EnvironmentAuthenticatorType: String {
+    case IAM = "iam"
+    case basic = "basic"
+    case CP4D = "cp4d"
+    case noAuth = "noauth"
+    case bearerToken = "bearerToken"
+}
+
+@available (iOS, unavailable, message: "ConfigBasedAuthenticatorFactory is currently available on Linux only.")
+@available (tvOS, unavailable, message: "ConfigBasedAuthenticatorFactory is currently available on Linux only.")
+@available (iOSMac, unavailable, message: "ConfigBasedAuthenticatorFactory is currently available on Linux only.")
+@available (watchOS, unavailable, message: "ConfigBasedAuthenticatorFactory is currently available on Linux only.")
+public struct ConfigBasedAuthenticatorFactory {
+    @available(iOS 9.0, *)
+    static public func getAuthenticator(credentialPrefix: String) throws -> Authenticator {
```

Comment:
That was inserted in by XCode, but I think I need to go back and modify the deployment targets to make sure that it is implied that anything below ios 9 is not gonna work.

XCode complains about it because Swift `@available` annotations still require the file to build on XCode, vs `#if os(Linux)` which avoids building it.